pkg/status/combinedstatus.go

```diff
@@ -83,6 +87,13 @@ func (c *Controller) updateOrCreateCombinedStatus(ctx context.Context, combinedS
 	// set results
 	combinedStatus.Results = generatedCombinedStatus.Results
 
+	if combinedStatus.Namespace == metav1.NamespaceNone {
+		combinedStatus.Namespace = clusterScopedObjectsNamespace
+		if err := c.ensureNamespaceExists(ctx, combinedStatus.Namespace); err != nil {
+			return fmt.Errorf("failed to ensure namespace exists: %w", err)
+		}
+	}
```

Comment:
I suspect that the code will be simpler to understand if `combinedStatus.Namespace` never holds the wrong value --- i.e., the construction in `func (c *combinedStatusResolution) generateCombinedStatus` always puts the right value there.